### PR TITLE
add GPIO defines for C3

### PR DESCRIPTION
**Pull request by NiklasVoigt**
Problem: https://github.com/espressif/arduino-esp32/issues/5502#issuecomment-984592239
Solution: https://github.com/tasmota/arduino-esp32/pull/31 from @Jason2866

### CODE REVIEW

**Comment by GYSS1204** [comment]
Hello, I changed the code to the code you updated, but the problem still appears.

**Comment by Jason2866** [comment]
@GYSS1204  Try this complete Arduino framework. It is builded on latest IDF4.4 and latest Arduino master including this change. It does work with my C3.
Platformio setup
```
platform                    = https://github.com/tasmota/platform-espressif32.git#feature/arduino-idf-v4.4
platform_packages           = framework-arduinoespressif32 @ https://github.com/Jason2866/esp32-arduino-lib-builder/releases/download/532/framework-arduinoespressif32-release_v4.4-c3ff20355.tar.gz
```

**Comment by me-no-dev** [comment]
I am a bit confused how this change helps (apart from maybe the adc init). For C3 `USE_ESP_IDF_GPIO` is defined as `1`, so it never uses the mux table.

**Comment by atanisoft** [comment]
@me-no-dev Perhaps it would simplify it if all architectures defined `USE_ESP_IDF_GPIO`

**Comment by me-no-dev** [comment]
@atanisoft that will have to be tested once we move all other peripherals to IDF API. It is the end goal though :)

File: cores/esp32/esp32-hal-gpio.c [hunk 1]
```diff
@@ -192,7 +221,7 @@ extern void ARDUINO_ISR_ATTR __pinMode(uint8_t pin, uint8_t mode)
 #endif
 	} else if(mode == ANALOG){
 #if !CONFIG_IDF_TARGET_ESP32C3
-		//adc_gpio_init(ADC_UNIT_1, ADC_CHANNEL_0);
+		adc_gpio_init(ADC_UNIT_1, ADC_CHANNEL_0);
```
Comment (P-R-O-C-H-Y) [inline cores/esp32/esp32-hal-gpio.c:224]: Is there any reason for uncommenting this line?

**Pull request merged**